### PR TITLE
Add recurring donations module

### PR DESCRIPTION
**Pull request by jonasnick**
Allows to send tallyco.in donations in given intervals with clightning by adding something like this to your `configuration.nix`:
```
services.recurring-donations.enable = true;
# Specify the receivers of the donations. By default donations are every
# Monday at a randomized time.
services.recurring-donations.tallycoin = {
  "djbooth007" = 1000;
  "hillebrandmax" = 1000;
};
```

### CODE REVIEW

File: modules/recurring-donations.nix [hunk 1]
```diff
@@ -0,0 +1,108 @@
...
+      NAME=$1
+      AMOUNT=$2
+      echo Attempting to pay $AMOUNT sat to $NAME
+      INVOICE=$(torsocks curl -d "satoshi_amount=$AMOUNT&payment_method=ln&id=$NAME&type=profile" -X POST https://api.tallyco.in/v1/payment/request/ | jq -r '.lightning_pay_request') 2> /dev/null
+      if [ -z "$INVOICE" ] || [ "$INVOICE" = "null" ]; then
+        echo "ERROR: did not get invoice from tallycoin"
+        return
+      fi
+      # Decode invoice and compare amount with requested amount
+      DECODED_AMOUNT=$($LNCLI decodepay "$INVOICE" | jq -r '.amount_msat' | head -c 4)
```
Comment (nixbitcoin) [inline modules/recurring-donations.nix]: I'm getting the following error when entering amounts below 1000, for example 500:

```
Apr 19 16:31:02 nix-bitcoin systemd[1]: Starting Run recurring-donations...
Apr 19 16:31:02 nix-bitcoin bash[304]: Attempting to pay 500 sat to djbooth007
Apr 19 16:31:02 nix-bitcoin bash[304]:   % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
Apr 19 16:31:02 nix-bitcoin bash[304]:                                  Dload  Upload   Total   Spent    Left  Speed
Apr 19 16:31:17 nix-bitcoin bash[304]: [1.3K blob data]
Apr 19 16:31:17 nix-bitcoin bash[304]: ERROR: requested amount and invoice amount do not match. 500 vs 5000
Apr 19 16:31:17 nix-bitcoin bash[304]: Attempting to pay 500 sat to hillebrandmax
Apr 19 16:31:17 nix-bitcoin bash[304]:   % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
Apr 19 16:31:17 nix-bitcoin bash[304]:                                  Dload  Upload   Total   Spent    Left  Speed
Apr 19 16:31:23 nix-bitcoin bash[304]: [553B blob data]
Apr 19 16:31:23 nix-bitcoin bash[304]: ERROR: requested amount and invoice amount do not match. 500 vs 5000
```

**Comment by nixbitcoin** [comment]
Please see https://github.com/jonasnick/nix-bitcoin/pull/41 for updated `docs/usage.md` documentation for this PR.

**Comment by nixbitcoin** [comment]
https://github.com/jonasnick/nix-bitcoin/issues/35

File: modules/recurring-donations.nix (hunk 1 above)
Comment (jonasnick) [reply modules/recurring-donations.nix]: Yeah, the way the amount is read out from the invoice is totally wrong. Fixing...

**Comment by jonasnick** [comment]
Fixed amount check bug

File: modules/recurring-donations.nix [hunk 2]
```diff
@@ -0,0 +1,108 @@
...
+
+  config = mkIf cfg.enable {
+    systemd.services.recurring-donations = {
+      description = "Run recurring-donations";
+      requires = [ "clightning.service" ];
+      after = [ "clightning.service" ];
+      path  = [ pkgs.clightning pkgs.curl pkgs.torsocks pkgs.sudo pkgs.jq ];
+      serviceConfig = {
+        ExecStart = "${pkgs.bash}/bin/bash ${recurring-donations-script}";
+        # TODO: would be better if this was operator, but I don't get sudo
```
Comment (nixbitcoin) [inline modules/recurring-donations.nix]: I'm getting `sudo: /nix/store/fm9zn0zvcsbj83d0qdshpvq049c1qk76-sudo-1.8.24/bin/sudo must be owned by uid 0 and have the setuid bit set` when trying to run with `LNCLI="sudo -u clightning lightning-cli --lightning-dir=${config.services.clightning.dataDir}"` and `User = "operator";`. Are you getting a similar error? Is this an upstream bug?

Comment (jonasnick) [reply modules/recurring-donations.nix]: Yes I was getting the same error

**Pull request merged**